app/dose-event-slider/dose-event-slider.component.html

```diff
@@ -1,21 +1,30 @@
-<ion-item-sliding *ngIf="doseEvent">
+<ion-item-sliding #slidingItem *ngIf="doseEvent">
 
     <ion-item>
         <ion-avatar item-left>
-            <ion-icon name="checkmark-circle" primary></ion-icon>
+            <ion-icon name={{doseEvent.action}} primary></ion-icon>
```

Comment:
A couple things here:

1. Even when you use the mustaches, you still need to surround them with quotes to be valid HTML:  `{{doseEvent.action}}`.

2. The proper way to do this in Angular 2 is with property binding, which uses brackets around the property name, and then just the name of the variable as the value. An example of this is shown on the Ionic documentation for `ion-icon`  (http://ionicframework.com/docs/v2/components/#icons).

This would be

```html
<ion-icon [name]="doseEvent.action"
          primary ></ion-icon>
```